spec/generators/rspec/scaffold/scaffold_generator_spec.rb

```diff
@@ -104,7 +104,7 @@
     before  { run_generator %w[admin/posts] }
     it { is_expected.to exist }
     it { is_expected.to contain(/^RSpec.describe "\/admin\/posts", #{type_metatag(:request)}/) }
-    it { is_expected.to contain('admin_post_url(admin_post)') }
+    it { is_expected.to contain('admin_post_url(post)') }
```

Comment:
But why?

```ruby
      accounts_user = Accounts::User.create! valid_attributes
      get accounts_user_url(accounts_user), as: :json
```

It makes it clear what namespace a certain model belongs to.
`Accounting::User` and `Reader::User` might be quite different.

If RSpec is used as a project documentation tool, I'd prefer it to be more specific by default.

What do you think?

Reply:
Before my changes the generator output was:
```ruby
# Output
user = Accounts::User.create! valid_attributes
get accounts_user_url(accounts_user), as: :json

# Source
# <%= file_name #>  returns 'user'
<%= file_name %> = <%= class_name %>.create! valid_attributes
get <%= show_helper %>, as: :json
```

To keep the pattern from another templates, example [controller_spec](https://github.com/rspec/rspec-rails/blob/efdc7b5cd9f4d7a0dba409a730609fd41585522d/lib/generators/rspec/scaffold/templates/controller_spec.rb#L57) ,i have use the file_name to be argument from path. In my pratice i use more the small name like user, but thinking as project documentation tool is better keep the full_name from resource, do you like that i change all `file_name` calls from generator source to be [ns_file_name](https://github.com/rspec/rspec-rails/blob/efdc7b5cd9f4d7a0dba409a730609fd41585522d/lib/generators/rspec/scaffold/scaffold_generator.rb#L79)?

Reply:
@JonRowe @benoittgt What do you guys think? `accounts_user` or just `user` for `Accounts::User`?

Reply:
They should be the same and the spec should work either way?

Reply:
Yes. I explained the problem with more details in this issue #2491 . We can fix the spec generator to use `accounts_user` or just `user`, in my PR, I chose `user`.
In order to change to `accounts_user` and keep the pattern from another templates generators, we need to change some other lines.